components/brave_new_tab_ui/apiEventsToStore.ts

```diff
@@ -43,8 +49,9 @@ export function wireApiEventsToStore () {
     }
     binanceInitData()
     getActions().setInitialData(initialData)
-    getActions().setFirstRenderGridSitesData(initialData)
     // Listen for API changes and dispatch to store
+    topSitesAPI.updateMostVisitedInfo()
```

Comment:
Doesn't seem like this needs to happen in the promise handler for `getInitialData`? Could it go on line 55 after the call to add the change listener (since it is an awkward call to fire the same change listener)?

Reply:
Fixed